pallets/teerex/ias-verify/src/lib.rs

```diff
@@ -216,6 +286,114 @@ pub fn verify_ias_report(cert_der: &[u8]) -> Result<SgxReport, &'static str> {
 	parse_report(netscape.attestation_raw)
 }
 
+pub fn parse_ias_report(cert_der: &[u8]) -> Result<SgxEnclaveMetadata, &'static str> {
+	#[cfg(test)]
+	println!("verifyRA: start verifying RA cert");
+
+	let cert = CertDer(cert_der);
+	let netscape = NetscapeComment::try_from(cert)?;
+	let sig_cert = webpki::EndEntityCert::from(&netscape.sig_cert).map_err(|_| "Bad der")?;
+
+	verify_signature(&sig_cert, netscape.attestation_raw, &netscape.sig)?;
+
+	// FIXME: now hardcoded. but certificate renewal would have to be done manually anyway...
+	// chain wasm update or by some sudo call
+	let valid_until = webpki::Time::from_seconds_since_unix_epoch(1573419050);
+	verify_server_cert(&sig_cert, valid_until)?;
+
+	let quote_add = netscape.quote_add.unwrap();
+	let quote_inputs = quote_add.quote_inputs;
+	let (report_inputs, quote_status, quote_body) = parse_sgx_quote(netscape.attestation_raw)?;
+
+	Ok(SgxEnclaveMetadata { report_inputs, quote_inputs, quote_status, quote_body })
+}
+
+fn parse_sgx_quote(
+	report_raw: &[u8],
+) -> Result<(SgxReportInputs, SgxStatus, Vec<u8>), &'static str> {
+	// parse attestation report
+	let attn_report: Value = match serde_json::from_slice(report_raw) {
+		Ok(report) => report,
+		Err(_) => return Err("RA report parsing error"),
+	};
```

Comment:
just use `map_err`?

Reply:
we can use a specified struct instead of `Value` to parse the `report_raw` to avoid parsing the json value like below.